src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3405,12 +3534,44 @@ tx_gateway_frontend::get_all_transactions() {
     auto ntp_meta = _metadata_cache.local().get_topic_metadata(
       model::tx_manager_nt);
     if (!ntp_meta) {
-        vlog(
-          txlog.error,
-          "Topic {} doesn't exist in metadata cache",
+        if (!co_await try_create_tx_topic()) {
+            co_return tx_errc::partition_not_exists;
+        }
+        try {
+            auto ec = co_await _controller->get_api().local().wait_for_topic(
+              model::tx_manager_nt,
+              topic_wait_timeout + model::timeout_clock::now());
+            if (ec) {
+                vlog(
+                  txlog.warn,
+                  "Error waiting for transaction manager topic "
+                  "{} to be created - {}",
+                  model::tx_manager_nt,
+                  ec);
+                // topic is creating, reply with not_coordinator error fot
+                // the client to retry
+                co_return tx_errc::partition_not_exists;
+            }
+        } catch (const ss::timed_out_error& e) {
+            vlog(
+              txlog.warn,
+              "Error waiting for transaction manager topic {} to be created - "
+              "{}",
+              model::tx_manager_nt,
+              e);
+        }
+
+        ntp_meta = _metadata_cache.local().get_topic_metadata(
           model::tx_manager_nt);
-        co_return tx_errc::partition_not_exists;
+        if (!ntp_meta) {
+            vlog(
+              txlog.error,
+              "Transaction manager topic {} not found",
+              model::tx_manager_nt);
+            co_return tx_errc::partition_not_exists;
+        }
```

Comment:
nit: I think this could be factored into a helper, similar logic in find_coordinator.

Reply:
done